tests/Routing/RouteRegistrarTest.php

```diff
@@ -1088,6 +1088,68 @@ public function testCanRemoveMiddlewareFromGroupUnregisteredGroup()
         $this->assertEquals([], $this->router->getMiddlewareGroups());
     }
 
+    public function testCanRegisterApiSingleton()
+    {
+        $this->router->apiSingleton('users', RouteRegistrarControllerStub::class);
+
+        $this->assertCount(2, $this->router->getRoutes());
+
+        $this->assertTrue($this->router->getRoutes()->hasNamedRoute('users.show'));
+        $this->assertTrue($this->router->getRoutes()->hasNamedRoute('users.update'));
+    }
+
+    public function testCanRegisterCreatableApiSingleton()
+    {
+        $this->router->apiSingleton('users', RouteRegistrarControllerStub::class)->creatable();
+
+        $this->assertCount(4, $this->router->getRoutes());
+
+        $this->assertTrue($this->router->getRoutes()->hasNamedRoute('users.store'));
```

Comment:
I would assert `users.show` and `users.update` here as well (also applies to some of the tests below this one)